joku/cogs/games/stocks.py

```diff
@@ -175,6 +175,94 @@ def _identify_stock(self, channels: typing.Sequence[discord.abc.GuildChannel], n
 
         await self.bot.redis.increase_history_count(message.channel)
 
+    @commands.group(pass_context=True, invoke_without_command=True, aliases=["shares", "share", "asset"])
+    async def assests(self, ctx: Context, *, target: discord.Member = None):
+        """
+        Gets the current amount of § a user has in assests.
+
+        If no target is provided, it will show your balance.
+        """
+        user = target or ctx.message.author
+        if user.bot:
+            await ctx.channel.send(":x: Bots cannot earn money.")
+            return
+
+        currency = await ctx.bot.database.get_user_currency(user)
+
+        guild = await ctx.bot.database.get_or_create_guild(ctx.guild)
+        if not guild.stocks_enabled:
+            await ctx.send(":x: Stocks are not enabled for this server.")
+            return
+
+        #Display only Assets the user owns (stolen from currency)
+        if ctx.channel.permissions_for(ctx.guild.me).embed_links:
+            day = self.rng.randint(1, 31)
+            hour, minute, second = self.rng.randint(0, 23), self.rng.randint(0, 59), self.rng.randint(0, 59)
+
+            ts = datetime.datetime(year=2008, month=7, day=day, hour=hour, minute=minute, second=second)
+
+            em = discord.Embed(title="First National Bank of Joku")
+            em.description = "You asset value is based on best estimates from current market prices."
+            em.set_author(name=user.display_name)
+
+            stocks = await ctx.bot.database.get_user_stocks(user)
+            em.add_field(name="Shares held", value=sum(userstock.amount for userstock in stocks))
+            em.add_field(name="Asset values", value="§{:.2f}".format(sum(userstock.amount * userstock.stock.price for
+                                                                         userstock in stocks)))
+
+            em.set_thumbnail(url=user.avatar_url)
+            em.timestamp = ts
+            em.set_footer(text="Thanks capitalism!")
+            em.colour = user.colour
+            await ctx.send(embed=em)
+        else:
+            await ctx.channel.send("User **{}** has `§{}`.".format(user, currency))
+
+    @assets.command(pass_context=True, aliases=["top", "leaderboard"])
+    async def richest(self, ctx: Context, *, what: str = "value"):
+        """
+        Shows the top 10 stock holders in this guild by asset worth
+        """
+        guild = await ctx.bot.database.get_or_create_guild(ctx.guild)
+        if not guild.stocks_enabled:
+            await ctx.send(":x: Stocks are not enabled for this server.")
+            return
+        if what == "worth": #order by amount of stock owned
+            users = await ctx.bot.database.get_multiple_users(*ctx.message.guild.members, 
+                                                           order_by=sum(userstock.amount * userstock.stock.price for 
```

Comment:
I don't think this will actually work, if I know SQLAlchemy.

Also, `stocks` would be undefined.